Python/compile.c

```diff
@@ -8615,7 +8620,7 @@ optimize_basic_block(struct compiler *c, basicblock *bb, PyObject *consts)
         struct instr *inst = &bb->b_instr[i];
         int oparg = inst->i_oparg;
         int nextop = i+1 < bb->b_iused ? bb->b_instr[i+1].i_opcode : 0;
-        if (is_jump(inst)) {
+        if (is_jump(inst) || is_block_push(inst)) {
```

Comment:
Hm, I didn't realize virtual opcodes made it this far.

Is there any value to including them in the jump threading prep here, though? The peephole optimizer doesn't know how to handle them anyways...

Reply:
Yes, this is very early after codegen, and before the exception table is constructed.

This loop jumps over empty blocks at the target of a jump, I think it (theoretically) could happen with a setup-X.

Reply:
Ah, disregard my comment. I forgot that we were actually updating `inst->i_target` here!